File: leetCode/C++/263-uglyNumber.cc
#include <iostream>

using namespace std;

class Solution {
public:
    bool isUgly(int num) {
        if (num < 1)
            return false;
        int ugly_nums[] = {2, 3, 5};
        for (const auto &ugly_num: ugly_nums) {
            while (num % ugly_num == 0)
                num /= ugly_num;
        }
        return num == 1;
    }
};

int main(void) {
    Solution s;
    cout << s.isUgly(1) << endl;
    cout << s.isUgly(6) << endl;
    cout << s.isUgly(8) << endl;
    cout << s.isUgly(120) << endl;
    cout << s.isUgly(14) << endl;
    cout << s.isUgly(13) << endl;
    return 0;
};

File: leetCode/Python2/263-uglyNumber.py
class Solution(object):
    def isUgly(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num < 1:
            return False
        for i in [2, 3, 5]:
            while num % i == 0:
                num /= i
        return num == 1

if __name__ == "__main__":
     s = Solution()
     print s.isUgly(30)
     print s.isUgly(10)
     print s.isUgly(14)

